DQM/HcalCommon/interface/Constants.h

```diff
@@ -27,29 +27,71 @@ namespace hcaldqm
 		//	use conversion functions in Utilities.h
 		//	For fast look up
 		//	This is for uTCA Crates/FEDs only - no other way...
-		int const FED_uTCA_MAX_REAL = 50;
-		uint16_t const FED2CRATE[FED_uTCA_MAX_REAL] = {
-			24, 0, 20, 0, 21, 0, 25, 0, 31, 0,
-			35, 0, 37, 0, 34, 0, 30, 0, 22,22,
-			29,29, 32,32, 0, 0, 0, 0, 0, 0,
-			0, 0, 38, 0, 0, 0, 0, 0, 0, 0,
-			0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+		std::map<unsigned int, unsigned int> const crate2fed_map = {
+			{24, 1100},
+			{20, 1102},
+			{21, 1104},
+			{25, 1106},
+			{31, 1108},
+			{35, 1110},
+			{37, 1112},
+			{34, 1114},
+			{30, 1116},
+			{22, 1118},
+			{29, 1120},
+			{32, 1122},
+			{38, 1134},
+			{3, 724},
+			{7, 726},
+			{6, 728},
+			{13, 730},
 		};
-		uint16_t const CRATE2FED[50] = {
-			// 2017 values:
-			0,0,0,0,0,0,0,0,0,0,
-			0,0,0,0,0,0,0,0,0,0,
-			1102,1104,1118,0,1100,1106,0,0,0,1120,
-			1116,1108,1122,0,1114,1110,0,1112,1132,0,
-			0,0,0,0,0,0,0,0,0,0
-			// 2016 values:
-			//0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			//0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			//1102, 1104, 1118, 0, 1100, 1106, 0, 0, 0, 1120,
-			//1116, 1108, 1122, 0, 1114, 1110, 1132, 1112, 0, 0,
-			//0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+
+		std::map<unsigned int, unsigned int> const fed2crate_map = {
+			{724, 3},
+			{725, 3},
+			{726, 7},
+			{727, 7},
+			{728, 6},
+			{729, 6},
+			{730, 13},
+			{731, 13},
+			{1100, 24},
+			{1101, 24},
+			{1102, 20},
+			{1103, 20},
+			{1104, 21},
+			{1105, 21},
+			{1106, 25},
+			{1107, 25},
+			{1108, 31},
+			{1109, 31},
+			{1110, 35},
+			{1111, 35},
+			{1112, 37},
+			{1113, 37},
+			{1114, 34},
+			{1115, 34},
+			{1116, 30},
+			{1117, 30},
+			{1118, 22},
+			{1119, 22},
+			{1120, 29},
+			{1121, 29},
+			{1122, 32},
+			{1123, 32},
+			{1134, 38},
+			{1135, 38},	
 		};
 
+		std::vector<unsigned int> const fedList = {724,725,726,727,728,729,730,731,1100,1101,1102,1103,1104,1105,1106,1107,1108,1109,1110,1111,1112,1113,1114,1115,1116,1117,1118,1119,1120,1121,1122,1123,1134,1135};
```

Comment:
Please correct me if I'm wrong, but FEDNumbering.h doesn't seem to have what we need... it defines a much larger range than what we actually use, 
`     MINHCALFEDID = 700,
     MAXHCALFEDID = 731,
...
     MINHCALuTCAFEDID = 1100,
     MAXHCALuTCAFEDID = 1199,
`
Since the FED numbers are arbitrary (i.e. non-consecutive, chosen at the whims of the operations people), it seems to me that we can't get around typing out the list of FEDs.